packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts

```diff
@@ -128,7 +180,82 @@ describe('TransferDomain', () => {
       await expect(promise).rejects.toThrow('Src address must be a legacy or Bech32 address in case of "DVM" domain')
     })
 
-    it('(evm -> dvm) should fail if source address and source domain are not match', async () => {
+    it('(evm -> dvm) should fail if dst address is not legacy or Bech32 address in case of "DVM domain', async () => {
+      const addressP2SH = await container.getNewAddress('address', 'p2sh-segwit')
+      // initially transfer DFI
+      await client.account.transferDomain([
+        {
+          src: {
+            address: dvmAddr, // invalid
+            amount: '1@DFI',
+            domain: TransferDomainType.DVM
+          },
+          dst: {
+            address: evmAddr,
+            amount: '1@DFI',
+            domain: TransferDomainType.EVM
+          }
+        }
+      ])
+      await container.generate(1)
+
+      const promise = client.account.transferDomain([
+        {
+          src: {
+            address: evmAddr,
+            amount: '1@DFI',
+            domain: TransferDomainType.EVM
+          },
+          dst: {
+            address: addressP2SH, // <- non legacy or Bech32 addres
+            amount: '1@DFI',
+            domain: TransferDomainType.DVM
+          }
+        }
+      ])
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow('Dst address must be a legacy or Bech32 address in case of "DVM" domain')
+      await container.generate(1)
+
+      // revert transfer
```

Comment:
hmm.. why need revert here tho?? 🤔

Reply:
There are testcase below where we are checking balance of evmAddress (considering there is no balance on evmaddress). couple of rows above we transferred 1 DFI so that we get exact error message `Dst address must be a legacy or Bech32 address in case of "DVM" domain` , to make it even I reverted the txn

Reply:
resolved